libmusic/include/lmtypes.h

```diff
@@ -145,22 +159,47 @@ typedef struct Chord {
                 ((c1.__mRootNote == c2.__mRootNote) && (c1.__mQuality < c2.__mQuality)));
     }
 
-    friend std::ostream& operator<<(std::ostream& os, const Chord &c)
+    std::string toHarte() const
     {
-        os << c.__mRootNote << c.__mQuality;
-        if (c.__mBassNote != note_Unknown)
-            os << "/" << c.__mBassNote;
-        return os;
+        std::ostringstream ss;
+
+        ss << __mRootNote;
+        if (__mRootNote != note_Unknown) {
+            ss << ":" << __mQuality;
+        }
+        if (__mBassInterval > 1) {
+            ss << "/" << (int)__mBassInterval;
```

Comment:
is this cast to `int` mandatory?

Reply:
yes. Apparently `<<` is not overloaded for `int8_t` which leads to empty output as a result.

Reply:
Turns out `int8_t` is in fact `signed char`, so `<<` attempts to print un-printable characters.
Why do we have to use this strange `int8_t` anyway? Can't we just use plain `int` and don't worry about casting each time we use it?

Reply:
That's probably my habit from the embedded world. Built in types with their variable width greatly impact code portability.